src/workerd/io/worker.c++

```diff
@@ -1626,6 +1628,7 @@ Worker::Worker(kj::Own<const Script> scriptParam,
             if (script->isModular()) {
               // Use `env` variable.
               bindingsScope = v8::Object::New(lock.v8Isolate);
+              lock.setWorkerEnv(lock.v8Ref(bindingsScope.As<v8::Value>()));
```

Comment:
Could we check the compat flag here, and not even call `setWorkerEnv()` (or call it with an empty object) if the flag is set? Rather than checking it on every access.

It doesn't make a difference currently, but if we start adding other ways to get an env ambiently, the compat flag applies to all of them, which is what I'd want as a user of this flag.

Reply:
If you look further down, you'll see that I've changed from storing `env` in `Worker::Impl` and instead store it in the lock here, so this sets it for the existing uses also, replacing the `impl->env = ` line was down below.

Reply:
If we end up adding other ways to get to the env, then we can move the code currently in `EnvModule::getCurrent()` into a more central location since that does everything that is needed

Reply:
Wait, I think this breaks dynamic bindings: There can be multiple `Worker` objects in the same isolate, each with a different `env`, if they have bindings that change from request to request. Dynamic bindings were only ever exposed internally and we want to get rid of them, but a lot of internal workers are still relying on them today, so if you start storing the `env` on the isolate, concurrent requests will stomp each other.

Reply:
Note that we don't need imported env to support dynamic bindings. For people using dynamic bindings, we can just say: don't use imported env. So I think imported env can just reflect the _first_ env ever received. But the env delivered to event handlers as a parameter still needs to be the per-request env, which we still need to store in `Worker::Impl`.

Reply:
hmm... I think there's a likelihood of weird bugs with that approach... let me play with it a bit to see what approaches may work.